### ISSUE TITLE
MLFlow hosted in AWS cannot write to a different bucket?

### ISSUE THREAD

**Issue by daimonie** [mlflow/mlflow#4584]
In my setup, the MLFlow service is hosted in the AWS cloud. It has a bucket, `s3://***-mlflow/` where it stores its own files. However, that bucket is not the bucket where the service/devops team wants me to actually save the artifacts. However, whenever I try to use the experiment creation to change the S3 bucket where the files are changed, I run into issues.

A fragment of code:

    client = MlflowClient(
        tracking_uri=client_address
    )
    experiment_obj = client.get_experiment_by_name(
        experiment_name
    )
    experiment_id = experiment_obj.experiment_id
    mlflow.set_experiment(experiment_name)
    for experiment in client.list_experiments():
        exp_info = str(experiment)
        logging.info(f"\t-{exp_info}")
    with mlflow.start_run(
            run_name=run_name,
            experiment_id=experiment_id
        ) as run:
        # It crashes before it gets here
        pass

This gives an error. Note that I have included the logging of `list_experiments` : 

    [train] 2021-07-21 15:33:13,537 - Setting up mlflow.
    [train] 2021-07-21 15:33:13,537 - Experiment name test_experiment.
    [train] 2021-07-21 15:33:13,537 - https://de-mlflow.***.***:5000
    [train] 2021-07-21 15:33:13,586 - Found experiment 54.
    [train] 2021-07-21 15:33:13,641 - Experiment test_experiment(54)
    [train] 2021-07-21 15:33:13,642 - Experiments found:
    [train] 2021-07-21 15:33:13,667 -   -<Experiment: artifact_location='s3://*-mlflow/mlflow-artifacts/0', experiment_id='0', lifecycle_stage='active', name='Default', tags={}>
    [train] 2021-07-21 15:33:13,668 -   -<Experiment: artifact_location='s3://*-projects/myteam', experiment_id='51', lifecycle_stage='active', name='test_model_sagemaker_v2_v2_v2_v2_v2_v2', tags={}>
    [train] 2021-07-21 15:33:13,668 -   -<Experiment: artifact_location='s3://*-projects/myteam', experiment_id='54', lifecycle_stage='active', name='test_experiment', tags={}>
    Traceback (most recent call last):
      File "train.py", line 125, in <module>
        train_and_deploy()
      File "train.py", line 77, in train_and_deploy
        experiment_id=experiment_id
      File "/usr/local/lib/python3.7/site-packages/mlflow/tracking/fluent.py", line 250, in start_run
        active_run_obj = client.create_run(experiment_id=exp_id_for_run, tags=tags)
      File "/usr/local/lib/python3.7/site-packages/mlflow/tracking/client.py", line 266, in create_run
        return self._tracking_client.create_run(experiment_id, start_time, tags)
      File "/usr/local/lib/python3.7/site-packages/mlflow/tracking/_tracking_service/client.py", line 96, in create_run
        tags=[RunTag(key, value) for (key, value) in tags.items()],
      File "/usr/local/lib/python3.7/site-packages/mlflow/store/tracking/file_store.py", line 482, in create_run
        experiment = self.get_experiment(experiment_id)
      File "/usr/local/lib/python3.7/site-packages/mlflow/store/tracking/file_store.py", line 363, in get_experiment
        experiment = self._get_experiment(experiment_id)
      File "/usr/local/lib/python3.7/site-packages/mlflow/store/tracking/file_store.py", line 334, in _get_experiment
        databricks_pb2.RESOURCE_DOES_NOT_EXIST,
    mlflow.exceptions.MlflowException: Could not find experiment with ID 54
    make: *** [train] Error 1

As you can see in the log-above, it found experiment `54`, which is set to use the `-projects` bucket, but throws a RESOURCE DOES NOT EXIST whenever I try to start a run with the experiment set.

**Comment by daimonie** [comment]
Resolved.

What seemed to have gone wrong is that `mlflow.start_run` is not aware of the `client` object. As in the code sample above, it wasn't aware of the `tracking_uri` (I think) and therefore couldn't find experiments, even though the experiment methods (relying on the `client` object) did find it. This seems to be because experiment tracking and model registry are developed separately but unsure. it would be good if we could e.g. `mlflow.set_client` and use it as a default when relying on an `MlflowClient` object.

Short version: Use both the client and mlflow.set_tracking_uri so both parts of mlflow are aware of the tracking_uri and use both start_run(experiment_id=... and mlflow.set_experiment so that both parts are aware.
